src/pj_ell_set.c

```diff
@@ -258,9 +258,8 @@ static int ellps_shape (PJ *P) {
         P->f = pj_atof (pj_param_value (par));
         if (HUGE_VAL==P->f)
             return proj_errno_set (P, PJD_ERR_INVALID_ARG);
-        if (0==P->f)
-            return proj_errno_set (P, PJD_ERR_INVALID_ARG);
-        P->rf = 1 / P->f;
+
+        P->rf = (P->f == 0) ? 1 : 1 / P->f;
```

Comment:
hum, actually in https://github.com/OSGeo/proj.4/blob/master/src/pj_ell_set.c#L547, it is set to HUGE_VAL

Reply:
Well, I'll just do the same then. I was just thinking what the logical value would be and ignoring the maths...

Reply:
Yes, the WKT convention is to let rf=0 mean f=0, but the WKT convention relates to textual representation, while here we have our fingers in the binary dough. 

The material in `int pj_calc_ellipsoid_params (PJ *P, double a, double es)` can be blamed on me, and the problem with adopting the WKT convention is that it we should be sure that it is accepted everywhere, including in the myriad of PJ_xxx.c files.

Rather than relying on this, I found it better to set rf to something that in IEEE arithmetic (where HUGE_VAL==Inf) would result in 1/rf = 0, and in non-IEEE (where HUGE_VAL==DBL_MAX), would at least give something very close to 0.

Obvoiusly, the WKT convention is sane, and one nice day with plenty of coffee and twice as much time, we should probably force it through to the binary level as well...